e2e/star-search.spec.ts

```diff
@@ -83,11 +87,38 @@ test("StarSearch OG image should exist", async ({ page }) => {
   await expect(page.locator('meta[name="twitter:card"]')).toHaveAttribute("content", "summary_large_image");
 });
 
-test("StarSearch OG image should exist for shared prompt", async ({ page }) => {
+test("StarSearch shared prompt", async ({ page }) => {
   await page.goto(
     "/star-search/?prompt=Show%20me%20the%20lottery%20factor%20for%20contributors%20in%20the%20remix-run/react-router%20project"
   );
 
+  expect(await page.getByRole("list", { name: "suggested prompts" })).not.toBeVisible();
+  expect(await page.getByRole("heading", { name: "StarSearch" })).not.toBeVisible();
+  expect(await page.getByRole("heading", { name: "Copilot, but for git history" })).not.toBeVisible();
+
+  const loginDialog = await page.getByRole("dialog", { name: "Login to try StarSearch", exact: true });
+  expect(loginDialog).toBeHidden();
+
+  const promptInput = await page.getByRole("textbox", { name: "Ask a question", exact: true });
+  expect(promptInput).toBeVisible();
+  expect(promptInput).toBeDisabled();
+
+  // get an element with role feed and name "StarSearch conversation" and check that it exists.
+  const feed = await page.getByRole("feed", { name: "StarSearch conversation" });
+  expect(feed).toBeVisible();
+  expect(feed).toHaveAttribute("aria-busy", "true");
+
+  const firstArticle = await feed.locator("article").first();
+  expect(await firstArticle.getByRole("heading", { name: "You" })).toBeVisible();
+  expect(await firstArticle.getByLabel("chat message")).toHaveText(
+    "Show me the lottery factor for contributors in the remix-run/react-router project"
+  );
+
+  expect(await feed.getByRole("progressbar", { name: "Loading..." })).toBeVisible();
+
+  // TODO: See if we can test the completion of the response since we don't want to add artificial delays.
```

Comment:
I'm leaving this as is for now. The chat stream coming in can take a while making the test long and I don't want to add artificial delays. We can mock the request for this as a means to test the flow of chat completing, loader disappearing and share icons appearing. We can make the assumption that the StarSearch stream endpoint is up and even if it wasn't that is a separate service.

Thoughts @open-sauced/engineering?

Reply:
It's a separate service entirely. I'd say as long as you can get to that page and send a message in that prompt field, test is 👍🏼 You might also consider _not_ sending a real `POST` since each of those requests to StarSearch cost us real money / compute. But, if there's no easy way to mock a `POST` in the e2e test, hitting the real service once in awhile is fine!

Reply:
Forgot about the cost as well. 😅 All roads lead to mocking this request. On it!